message/html/action/dislike.js

```diff
@@ -0,0 +1,51 @@
+var { h, computed, when } = require('mutant')
+var nest = require('depnest')
+
+exports.needs = nest({
+  'keys.sync.id': 'first',
+  'message.obs.dislikes': 'first',
+  'sbot.async.publish': 'first'
+})
+
+exports.gives = nest('message.html.action')
+
+exports.create = (api) => {
+  return nest('message.html.action', function dislike (msg) {
+    var id = api.keys.sync.id()
+    var disliked = computed([api.message.obs.dislikes(msg.key), id], doesNotLike)
+    return when(disliked,
+      h('a.undislike', {
+        href: '#',
+        'ev-click': () => publishDislike(msg, false)
+      }, 'Undislike'),
+      h('a.dislike', {
+        href: '#',
+        'ev-click': () => publishDislike(msg, true)
+      }, 'Dislike')
+    )
+  })
+
+  function publishDislike (msg, status = true) {
+    var dig = status ? {
+      type: 'vote',
+      channel: msg.value.content.channel,
+      vote: { link: msg.key, value: -1, expression: 'Shun' }
```

Comment:
why not `Dislike` and `Undislike`?

https://github.com/ssbc/patchwork/issues/486#issuecomment-292773977